packages/contracts-bedrock/contracts/L2/CrossDomainOwnable2.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Predeploys } from "../libraries/Predeploys.sol";
+import { L2CrossDomainMessenger } from "./L2CrossDomainMessenger.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title CrossDomainOwnable2
+ * @notice This contract extends the OpenZeppelin `Ownable` contract
+ *         for L2 contracts to be owned by contracts on L1. Note that
+ *         this contract is meant to be used with systems that use the
+ *         CrossDomainMessenger system. It will not work if the OptimismPortal
+ *         is used directly.
+ */
+abstract contract CrossDomainOwnable2 is Ownable {
+    L2CrossDomainMessenger internal messenger =
```

Comment:
Should this be immutable? There's no affordance made for modifying it, and the address is unlikely to change.

Reply:
I'll make it a constant, good call